sklearn/utils/_metadata_requests.py

```diff
@@ -963,7 +965,7 @@ def route_params(self, *, caller, params):
         -------
         params : Bunch
             A :class:`~utils.Bunch` of the form
-            ``{"object_name": {"method_name": {prop: value}}}`` which can be
+            ``{"object_name": {"method_name": {params: value}}}`` which can be
```

Comment:
I have noticed that `prop`, `props` and `params` are used interchangeably within the docs and variable names of the metadata routing files. I found this confusing and having future contributors in mind, I propose to go over code and docs to see if we could make everything into `params`, or even `metadata_params`(?). 

What do you think, @adrinjalali?

Reply:
`metadata_params` wouldn't make sense to me, but I'm okay with changing them to `params`.